### PR TITLE
Pkg fix

### PR DESCRIPTION
**Pull request by phanimarupaka**
This PR is to implement a new command, kpt pkg fix. This command reads the local package, modifies the package to use the latest kpt features and fixes any deprecated feature traces. This plays a key role in helping users to travel along with the new features in kpt.

### CODE REVIEW

File: internal/util/fix/fix.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		printFunc("created substitution with name %s", subst)
+	}
+	printFunc("created %d substitution in total", len(sfr.CreatedSubst))
+
+	for setter, err := range sfr.FailedSetters {
+		printFunc("failed to create setter with name %s: %v", setter, err)
+	}
+
+	for subst, err := range sfr.FailedSubst {
+		printFunc("failed to create setter with name %s: %v", subst, err)
```
Comment (prachirp) [inline internal/util/fix/fix.go:73]: Typo setter -> substitution

File: internal/util/fix/fix.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func (c Command) fixV1Setters() error {
+	printFunc := printFunc(c.StdOut, c.DryRun)
+	f := &fixsetters.SetterFixer{
+		PkgPath: c.PkgPath,
+		OpenAPIPath: filepath.Join(c.PkgPath, "Kptfile"),
+		DryRun:  c.DryRun,
+	}
+	sfr, err := f.FixV1Setters()
```
Comment (prachirp) [inline internal/util/fix/fix.go:52]: Missing error handling here

File: site/content/en/reference/pkg/fix/_index.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+type: docs
+description: >
+   Fix a local package.
+---
+<!--mdtogo:Short
+    Fix a local package.
+-->
+
+Fix reads the local package, modifies the package to use the latest kpt features 
+and fixes any deprecated feature traces
```
Comment (prachirp) [inline site/content/en/reference/pkg/fix/_index.md:13]: Use specific example, "and removes any deprecated feature traces like v1 setter comments"

File: site/content/en/reference/pkg/fix/_index.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+### Synopsis
+<!--mdtogo:Long-->
+```
+kpt pkg fix LOCAL_PKG_DIRECTORY [flags]
+
+LOCAL_PKG_DIRECTORY:
+  Local directory with kpt package. Directory must exist and
+  contain a Kptfile.
+
```
Comment (mortent) [inline site/content/en/reference/pkg/fix/_index.md:39]: Include information about the `dry-run` flag here.

File: site/content/en/reference/pkg/fix/_index.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Fix"
+linkTitle: "fix"
+type: docs
+description: >
+   Fix a local package.
+---
+<!--mdtogo:Short
+    Fix a local package.
```
Comment (mortent) [inline site/content/en/reference/pkg/fix/_index.md:9]: Can we expand a little bit here to make it clear what we mean by "fix"

File: internal/cmdfix/cmdfix.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Google LLC
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply internal/cmdfix/cmdfix.go:1]: Isn't it kpt license year which should be 2019 ? What about other places ?

File: internal/util/fix/fix.go (hunk 1 above)
Comment (phanimarupaka) [reply internal/util/fix/fix.go]: Done. Thanks.

File: internal/util/fix/fix.go (hunk 2 above)
Comment (phanimarupaka) [reply internal/util/fix/fix.go:52]: Good catch. Done and added tests.

File: site/content/en/reference/pkg/fix/_index.md (hunk 3 above)
Comment (phanimarupaka) [reply site/content/en/reference/pkg/fix/_index.md]: This goes into cli -h message as well. So better to leave it generic. We can have specific guides for each case if we need in future.

File: site/content/en/reference/pkg/fix/_index.md (hunk 4 above)
Comment (phanimarupaka) [reply site/content/en/reference/pkg/fix/_index.md:45]: Done.

File: site/content/en/reference/pkg/fix/_index.md (hunk 5 above)
Comment (phanimarupaka) [reply site/content/en/reference/pkg/fix/_index.md]: Done. Followed the short pattern like other commands, but expanded it  now.

**Pull request merged**